### ISSUE TITLE
tfrecordreader could also support SequenceExample protobuf

### ISSUE THREAD

**Issue by praateekmahajan** [vahidk/tfrecord#16]
Looks like `tfrecord_loader` [assumes](https://github.com/vahidk/tfrecord/blob/master/tfrecord/reader.py#L85) that the tfrecord files are of type `Example`. 

The drawback of Example is that it doesn't support Arrays of Arrays. (Which might be a common use case in ML world). One can get away with it for now by flattening the array while writing, and then PyTorch doing a `reshape(-1, DIM)`.

However it might be worth to add [SequenceExample](https://www.tensorflow.org/api_docs/python/tf/train/SequenceExample) which (seems) to be a superset of Example.

Looks like, we can even replace the Example with SequenceExample and everything should work the way it does, plus we might get new features with a little more effort.

Let's assume we have three features
1. `int_feature`     : Integer             :  `int64_list`
1. `arr_feature`     : Array(Float)        :  `float64_list`
1. `arr_arr_feature` : Array(Array(Float)) :  `float64_list`

Currently `arr_arr_feature` can not be accessed in the given framework. However below I go through an example if we use SequenceExample on how `arr_arr_feature` could be accessible.

```python       
# current code
example = example_pb2.Example()
example.ParseFromString(record)
example.features.feature['int_feature'].int64_list.value 
example.features.feature['arr_feature'].float64_list.value 
# not possible to access arr_arr_feature
```

can be replaced by

```python     
# suggested code  
example = example_pb2.SequenceExample()
example.ParseFromString(record)
 # replace .feature with .context
example.context.feature['int_feature'].int64_list.value 
example.context.feature['arr_feature'].float64_list.value
[ft.float_list.value for ft in example.feature_lists.feature_list['arr_arr_feature'].feature]
```
I'm not totally impressed with the way I've done  `[ft.float64.value for ft in example.feature_list.feature_list.....]`. 

Let me know if you like the idea, I can create a PR for the same.

More stuff here : 
1. https://www.tensorflow.org/api_docs/python/tf/train/SequenceExample
1. https://www.tensorflow.org/api_docs/python/tf/train/Example

**Comment by vahidk** [comment]
Other than reading existing sequence example datasets I don't see a lot benefits with supporting this. You can already flatten sequences and write it in Example format, then use a custom loader to form batches of sequences.

**Comment by alexi** [comment]
The biggest benefit would be automatic padding variable length sequence. It can be impressively space inefficient to do the padding in the stored data.

**Comment by praateekmahajan** [comment]
@alexi, we had the same issue for padding variable length sequences. One way to get away with that without having to pad in stored data is to leverage [transformations](https://github.com/vahidk/tfrecord#transforming-input).

**Comment by vahidk** [comment]
Seems like there's a WIP PR to resolve this:
https://github.com/vahidk/tfrecord/pull/44
